client/packages/common/src/ui/forms/JsonForms/useJsonForms.tsx

```diff
@@ -186,6 +186,10 @@ export const useJsonForms = (
     }
   }, [createDoc]);
 
+  useEffect(() => {
```

Comment:
the `documentName` being stored in state causes a problem if the calling page uses `undefined` and then sends the correct document name. have to update the internal state. 

perhaps it isn't required though - can it not just use the prop?